### DISCUSSION TITLE
Vanguard

### DISCUSSION THREAD

**Discussion by JoyAndSad** [wasiejen/Free-Snap-Tap#13] [Q&A]
Will Vanguard ban me for using this program???

**Comment by wasiejen** [comment]
Welcome and thanks for trying out this little program.

Vanguard is a pretty intrusive kernel mode anti-cheat software from what I've read about it.

> So can it recognise the use of the program? 
- My answer would be: easily.

This program is not meant to circumvent Anti Cheat Software (ACS), but is it even necessary?
All depends on how Snap Tap will be viewed - as cheat or just another way to optimise your input (e.g. like a keyboard :-) )?

Some earlier discussion to that topic: #11 

And up until now I have not seen or heard of any bans for using any form of Snap Tap.

The only thing close to it is the ban of null binds in the FACEIT ESEA League Matches - but hardware based Snap Tap remains still allowed there. (For context: null binds in cs2 are just some lines of code in the config file to prevent simultanous input of the a and d key) So there is still a hot discussion whether this is just an unfair advantage for anyone who can affort such hardware. In my opinion they just don't want to open the gates for more config file edits - and any tampering in it. If they consider Snap Tap a cheat in general, they would also ban the hardware for it.

So imho, up to this time it seems save to be used (maybe not in leagues), but that can always change and you should keep an eye open for news regarding Snap Tap and vanguard/riots stance on it. Most likely they will announce it if they change their view on it being a cheat, because it will influence their leagues also.
